content/docs/07-reference/ai-sdk-ui/02-use-completion.mdx

```diff
@@ -1,28 +1,181 @@
+      dark
+      prompt={false}
+    />
+  </Tab>
+</Tabs>
 
-<Snippet text={`import { useCompletion } from "ai/react"`} prompt={false} />
+## API Signature
 
-### Svelte
+### Parameters
 
-<Snippet text={`import { useCompletion } from "ai/svelte"`} prompt={false} />
+<PropertiesTable
+  content={[
+    {
+      name: 'api',
+      type: "string = '/api/completion'",
+      description: 'The text completion API endpoint offered by the provider.',
+    },
+    {
+      name: 'id',
+      type: 'string',
+      description:
+        'An unique identifier for the completion. If not provided, a random one will be generated. When provided, the `useCompletion` hook with the same `id` will have shared states across components. This is useful when you have multiple components showing the same chat stream',
+    },
+    {
+      name: 'initialInput',
+      type: 'string',
+      description: 'An optional string for the initial prompt input.',
+    },
+    {
+      name: 'initialCompletion',
+      type: 'string',
+      description: 'An optional string for the initial completion result.',
+    },
+    {
+      name: 'onResponse',
+      type: '(response: Response) => void',
+      description:
+        'An optional callback function that is called with the response from the API endpoint. Useful for throwing customized errors or logging.',
+    },
+    {
+      name: 'onFinish',
+      type: '(prompt: string, completion: string) => void',
+      description:
+        'An optional callback function that is called when the completion stream ends.',
+    },
+    {
+      name: 'onError',
+      type: '(error: Error) => void',
+      description:
+        'An optional callback that will be called when the chat stream encounters an error',
+    },
+    {
+      name: 'headers',
+      type: 'Record<string, string> | Headers',
+      description:
+        'An optional object of headers to be passed to the API endpoint.',
+    },
+    {
+      name: 'body',
+      type: 'any',
+      description:
+        'An optional, additional body object to be passed to the API endpoint.',
+    },
+    {
+      name: 'credentials',
+      type: "'omit' | 'same-origin' | 'include'",
+      description:
+        'An optional literal that sets the mode of credentials to be used on the request. Defaults to same-origin.',
+    },
+    {
+      name: 'sendExtraMessageFields',
+      type: 'boolean',
+      description:
+        "An optional boolean that determines whether to send extra fields you've added to `messages`. Defaults to `false` and only the `content` and `role` fields will be sent to the API endpoint.",
+    },
+    {
+      name: 'streamMode',
+      type: "'text' | 'stream-data'",
+      optional: true,
+      description:
+        'An optional literal that sets the mode of the stream to be used. Defaults to `stream-data`. If set to `text`, the stream will be treated as a text stream.',
+    },
+  ]}
+/>
 
-### Vue
+### Returns
```

Comment:
q: not blocking, i think for these reference pages, should we include the high level type that wraps these properties? e.g. for `useCompletion` = `UseCompletionHelpers`

Reply:
for `useChat` I've inlined it, might do the same here